src/core/xref.js

```diff
@@ -391,16 +390,22 @@ function showErrors({ errorsAmbiguous, errorsTermNotFound }) {
   }
 
   const titleForAmbiguous = "Error: Linking an ambiguous dfn.";
-  const hintForAmbiguous = `To disambiguate, you need to add a ${linkToDataCite} attribute.`;
-  for (const [term, data] of errorsAmbiguous) {
-    const specList = [...data.specs];
-    const count = specList.length;
-    const specs = specList.map(s => `**${s}**`).join(", ");
-    const msg =
-      `The term "**${term}**" is defined in ${count} ` +
-      `spec(s) in multiple ways, so it's ambiguous. ${hintForAmbiguous} ` +
-      `The specs where it's defined are: ${specs}.`;
-    showInlineError(data.elems, msg, titleForAmbiguous);
+  for (const { term, elems, results } of errorsAmbiguous.values()) {
+    const definedInSpecs = new Set(results.map(entry => entry.shortname));
+    const specs = [...definedInSpecs].map(s => `**${s}**`).join(", ");
+    const msg = `The term "**${term}**" is defined in ${specs} in multiple ways, so it's ambiguous.`;
+    let hint = "";
+    if (definedInSpecs.size === 1) {
+      // defined only in one spec but in multiple ways
+      const xrefFor = new Set([].concat(...results.map(entry => entry.for)));
+      const forContext = [...xrefFor].map(s => `**${s}**`).join(", ");
+      hint = `add \`data-xref-for\` attribute with value equal to one of the following: ${forContext}.`;
+    } else {
+      // defined in multiple specs
+      hint = `add ${dataCiteLink} attribute with value equal to one of the following: ${specs}.`;
```

Comment:
I wonder if we should mention that they should use the "[= for / term=]" syntax?

Reply:
Maybe add wiki link to `data-xref-for` and mention shorthand syntax there?

Reply:
Yeah, probably better

Reply:
We should move people off using data-*